File: java/src/main/java/com/fengtian/qianer/atcoder/abc/abc170/ABC170_C.java
package com.fengtian.qianer.atcoder.abc.abc170;

import java.util.Scanner;

public class ABC170_C {

    public static void main(String[] args) {
        final Scanner sc = new Scanner(System.in);

        final int x = sc.nextInt();
        final int n = sc.nextInt();

        final int[] p = new int[n];

        for (int i = 0; i < n; i++) {
            p[i] = sc.nextInt();
        }

        int diff = Integer.MAX_VALUE;
        int ans = 0;
        LOOP: for (int i = -1; i <= 101; i++) {
            for (int j = 0; j < n; j++) {
                if (p[j] == i) {
                    continue LOOP;
                }
            }
            if (Math.abs(x-i) < diff) {
                diff = Math.abs(x-i);
                ans = i;
            }
        }

        System.out.println(ans);
    }
}

File: python/src/abc/abc170/ABC_170_C.py
x, n = map(int, input().split())
p = list(map(int, input().split()))
n = list(i for i in range(-1, 110) if i not in p)

diff = 110
ans = 0

for i in n:
    if diff > abs(x-i):
        diff = abs(x-i)
        ans = i
        
print(ans)